operation.go

```diff
@@ -155,15 +155,36 @@ func (operation *Operation) ParseParamComment(commentLine string, astFile *ast.F
 
 	//five possible parameter types.
 	switch paramType {
+	case "[]query":
+		validType := TransToValidSchemeType(schemaType)
+		fmt.Println(IsPrimitiveType(validType))
+		if !IsPrimitiveType(validType) {
+			return fmt.Errorf("%s is not supported type for parameter %s", validType, paramType)
+		}
+		param = createParameter(paramType, description, name, validType, required)
 	case "query", "path", "header":
 		param = createParameter(paramType, description, name, TransToValidSchemeType(schemaType), required)
 	case "body":
-		param = createParameter(paramType, description, name, "object", required) // TODO: if Parameter types can be objects, but also primitives and arrays
-		if err := operation.registerSchemaType(schemaType, astFile); err != nil {
+		objType := "object"
+		if strings.HasPrefix(schemaType, "[]") {
```

Comment:
I know "it works" but this is not an elegant way.
You should do the job in  registerSchemaType because you have access to ast and you can identify what ever it is.